_pytest/fixtures.py

```diff
@@ -1068,6 +1068,9 @@ def parsefactories(self, node_or_obj, nodeid=NOTSET, unittest=False):
         self._holderobjseen.add(holderobj)
         autousenames = []
         for name in dir(holderobj):
+            # Skip properties before the following getattr executes them.
+            if isinstance(type(holderobj).__dict__.get(name), property):
```

Comment:
@bluetech basically any implementation of the descriptor protocol that does something at the class level can trigger

on the other hand i also think its fine to trigger those objects,  since well, python mantra is "consenting adults" - we dont need to protect people from their own traps too much